components/net_connect/eth_connect.c

```diff
@@ -0,0 +1,170 @@
+/*
+ * SPDX-FileCopyrightText: 2022-2025 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <string.h>
+#include "net_connect.h"
+#include "net_connect_private.h"
+#include "esp_event.h"
+#include "esp_eth.h"
+#include "ethernet_init.h"
+#include "esp_log.h"
+#include "esp_mac.h"
+#include "driver/gpio.h"
+#include "freertos/FreeRTOS.h"
+#include "freertos/task.h"
+#include "freertos/event_groups.h"
+
+
+static const char *TAG = "ethernet_connect";
+static SemaphoreHandle_t s_semph_get_ip_addrs = NULL;
+#if CONFIG_NET_CONNECT_IPV6
+static SemaphoreHandle_t s_semph_get_ip6_addrs = NULL;
```

Comment:
should the IPv6 be handled somehow globally, not per interface? Does it make sense to rework the architecture to provide netif-agnostic event/handlers, while keeping lifecycle netif/driver specific?